daemon/policy.go

```diff
@@ -305,7 +369,25 @@ func (d *Daemon) PolicyDelete(labels labels.LabelArray) (uint64, error) {
 		d.policy.Mutex.Unlock()
 		return rev, api.New(DeletePolicyNotFoundCode, "policy not found")
 	}
-	rev, deleted := d.policy.DeleteByLabelsLocked(labels)
+
+	// policySelectionWG is a waitgroup which is used to signal when the updating
+	// of all of the caches of allEndpoints in the rules which were added / updated
+	// have been updated.
+	var policySelectionWG sync.WaitGroup
+
+	// Get all endpoints at the time rules were added / updated so we can figure
+	// out which endpoints to regenerate / bump policy revision.
+	allEndpoints := endpointmanager.GetEndpoints()
+	epsIdentityConsumers := make([]policy.IdentityConsumer, len(allEndpoints))
+
+	// Need to explicitly convert endpoints to IdentityConsumers.
```

Comment:
Update comment with reference to https://github.com/golang/go/wiki/InterfaceSlice